modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1323,16 +1336,37 @@ private <T> CompletableFuture<T> continueResolvingByPk(
 
     }
 
+    /**
+     * Clean up txn locally.
+     *
+     * @param txId Tx ID.
+     * @param committed {@code True} if committed.
+     */
+    private void cleanupLocal(UUID txId, boolean committed) {
+        txCleanupReadyFutures.compute(txId, (id, txOps) -> {
```

Comment:
As was mentioned earlier I don't think that we need txnState for one-phase commit transactions.